### ISSUE TITLE
get_build_logs() fails with TypeError

### ISSUE THREAD

**Issue by pbabinca** [DBuildService/osbs#65]
Stripped traceback:
```
    build_log_contents = osbs_handler.get_build_logs(osbs_build_id)
  File "/usr/lib/python2.7/dist-packages/osbs/api.py", line 27, in catch_exceptions
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/osbs/api.py", line 203, in get_build_logs
    if build["status"] in ["Complete", "Failed"]:
TypeError: \'instancemethod\' object has no attribute \'__getitem__\'
'>
```
where `osbs_handler` is object created with:
```
from osbs.api import OSBS
from osbs.conf import Configuration
os_conf = Configuration()
build_conf = Configuration()
osbs_handler = OSBS(os_conf, build_conf)
```

I've tried to replace `build` with `build.json()` and this time it failed on `KeyError`:
```
  File "/usr/lib/python2.7/dist-packages/osbs/api.py", line 203, in get_build_logs
    return build.json()["metadata"]["labels"]["logs"]
KeyError: u'labels'
```
response from that
`build.json()` with pretty print:
```
{u'apiVersion': u'v1beta1',
 u'completionTimestamp': u'2015-05-12T11:40:41Z',
 u'duration': 0,
 u'kind': u'Build',
 u'metadata': {u'creationTimestamp': u'2015-05-12T11:40:27Z',
               u'name': u'docker-hello-world-20150512-114027',
               u'namespace': u'default',
               u'resourceVersion': u'192324',
               u'selfLink': u'/osapi/v1beta1/builds/docker-hello-world-20150512-114027?namespace=default',
               u'uid': u'af41762e-f89b-11e4-9e91-fa163e1012c3'},
 u'parameters': {u'output': {u'dockerImageReference': u'docker-registry.examle.com/pbabinca/docker-hello-world:20150512114027',
                             u'imageTag': u'pbabinca/docker-hello-world:20150512114027',
                             u'registry': u'docker-registry.example.com'},
                 u'resources': {},
                 u'source': {u'git': {u'ref': u'master',
                                      u'uri': u'git://git.example.com/users/pbabinca/docker-hello-world.git'},
                             u'type': u'Git'},
                 u'strategy': {u'customStrategy': {u'env': [{u'name': u'DOCK_PLUGINS',
                                                             u'value': u'{"prebuild_plugins": [{"name": "change_from_in_dockerfile"}, {"args": {"labels": {"Authoritative_Registry": "registry.example.com", "Vendor": "Foo bar, Inc.", "Architecture": "x86_64", "Build_Host": "foohost"}}, "name": "add_labels_in_dockerfile"}, {"args": {"command": "rhpkg sources"}, "name": "distgit_fetch_artefacts"}, {"args": {"repourls": []}, "name": "add_yum_repo_by_url"}, {"name": "inject_yum_repo"}, {"name": "dockerfile_content"}, {"args": {"registry_uri": "docker-registry.example.com", "insecure_registry": true}, "name": "change_source_registry"}], "postbuild_plugins": [{"args": {"registry_uri": "docker-registry.example.com", "insecure": true}, "name": "tag_by_labels"}, {"name": "tag_and_push"}, {"args": {"image_id": "BUILT_IMAGE_ID"}, "name": "all_rpm_packages"}, {"args": {"url": "http://10.3.11.31:9443/osapi/v1beta1/", "verify_ssl": false, "use_auth": false}, "name": "store_metadata_in_osv3"}, {"name": "remove_built_image"}]}'}],
                                                   u'exposeDockerSocket': True,
                                                   u'image': u'buildroot'},
                               u'type': u'Custom'}},
 u'status': u'Failed'}
```

**Comment by twaugh** [comment]
I touched this code last, I'm on it.

**Comment by twaugh** [comment]
It was still assuming the path was metadata/labels/logs, whereas it may now be metadata/annotations/logs.

I've updated the `fix-get_build_logs` branch to handle that.

**Comment by TomasTomecek** [comment]
Right! With new openshift we need to store everything in annotations, because labels have now strict validation. Thanks for looking into it, Tim.

**Comment by pbabinca** [comment]
fix-get_build_logs branch (b1ff87959b358957d6bf0add9665395f2842d36f) fixed my issue. Thanks!
